ui/v2.5/src/components/Scenes/SceneDetails/RatingSystem.tsx

```diff
@@ -0,0 +1,363 @@
+import React, { useState, useRef } from "react";
+import * as GQL from "src/core/generated-graphql";
+import { ConfigurationContext } from "src/hooks/Config";
+import Box from "@mui/material/Box";
+import StarIcon from "@mui/icons-material/Star";
+import StarBorderIcon from "@mui/icons-material/StarOutline";
+import { FormattedMessage, useIntl } from "react-intl";
+import { Col, Form, Row } from "react-bootstrap";
+import { FormUtils } from "src/utils";
+
+export interface IRatingSystemProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+  excludeLabel?: boolean;
+}
+
+export interface IRatingStarsProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+  precision: number;
+  maxRating: number;
+}
+
+export interface IRatingNumberProps {
+  value?: number;
+  onSetRating?: (value?: number) => void;
+  disabled?: boolean;
+}
+
+export const RatingNumber: React.FC<IRatingNumberProps> = (
+  props: IRatingNumberProps
+) => {
+  const [input, setInput] = useState<string | "0.0">();
+  const [previous, setPrevious] = useState<string | "0.0">();
+  const [useValidation, setValidation] = useState<boolean | true>();
+  function stepChange() {
+    setValidation(false);
+  }
+
+  function nonStepChange() {
+    setValidation(true);
+  }
+
+  function handleChange(e: React.ChangeEvent<HTMLInputElement>) {
+    let val = e.target.value;
+    if (!useValidation && props.onSetRating != null) {
+      e.target.value = Number(val).toFixed(1);
+      setInput(Number(val).toFixed(1));
+      setPrevious(Number(val).toFixed(1));
+      let tempVal = Number(val) * 10;
+      props.onSetRating(tempVal != 0 ? tempVal : undefined);
+      setValidation(true);
+      return;
+    }
+    const match = /(\d{0,1})(\d{0,1})(.?)((\d{0,1})?)/g.exec(val);
```

Comment:
Seems like this could be simplified to `(\d)(\d)(.?)((\d)?)`

Reply:
changed